RestaurantsNearMe/Data/APIKeyProvider.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  APIKey.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-11-09.
+//
+
+import Foundation
+
+@propertyWrapper
+struct APIKeyProvider {
+  private var key: APIKey?
+
+  var wrappedValue: APIKey? {
+    get {
+      return key
+    }
+    set {
+      // TODO update the envrionment with the new key
```

Comment:
this is incorrect, but I was trying to figure out how to leverage a property wrapper to store my API Key

Reply:
You could do something like this:

```
struct APIKey {
    enum Source {
        case cloudKit
    }

    let source: Source
    var value: String?

    var keyId: UUID {
        switch source {
        case .cloudKit:
            return UUID(uuidString: "975F2DC6-91AB-48CD-9775-3F17A4601F2D")!
        }
    }
}


@propertyWrapper
struct APIKeyProvider {
    var wrappedValue: APIKey

    /// Initializes an empty `APIKey` of type `wrappedValue`
    init(_ wrappedValue: APIKey.Source) {
        self.wrappedValue = APIKey(source: wrappedValue)
    }
}

final class MyViewModel {
    @APIKeyProvider(.cloudKit) var apiKey

    func fetchKeyIfNeeded() {
        guard apiKey.value == nil else {
            return
        }
        /// fetch API key for apiKey.keyId
        apiKey.value = <#newValue#>
    }
}
```